Data/ByteString.hs

```diff
@@ -1201,9 +1201,10 @@ findIndexEnd k (BS x l) = accursedUnutterablePerformIO $ withForeignPtr x $ \ f
 findIndices :: (Word8 -> Bool) -> ByteString -> [Int]
 findIndices p ps = loop 0 ps
    where
-     loop !n !qs | null qs           = []
-                 | p (unsafeHead qs) = n : loop (n+1) (unsafeTail qs)
-                 | otherwise         =     loop (n+1) (unsafeTail qs)
+     loop !n !qs = case findIndex p qs of
+                     Just !i -> (n+i) : loop (n+i+1) (unsafeDrop (i+1) qs)
```

Comment:
I'm curious whether it helps to avoid allocating a thunk here by evaluating the (n+i) also strictly, or, whether inlining generally takes of that, because callers will generally evaluate at least the first index strictly, and inlining will mean that the compiler will figure this out.  Still it might help to write:
```suggestion
                     Just !i -> let !j = n + i
                                 in j : loop (j+1) (unsafeDrop (i+1) qs)                     
```

Reply:
Done.